File: src/common/text_button.rb
require File.join(File.dirname(__FILE__), 'button')

class TextButton < Button
  def initialize(owner, view_options, callback_name, font, color = 0xffffffff, title = nil, &callback)
    super(owner, view_options, block_given? ? callback : callback_name)
    @font = font
    @title = title || callback_name.to_s.split(/_/).collect {|word| word.capitalize }.join(' ')
    @color = color
  end
  
  def render
    super
    @font.draw(@title, (@dx - @font.text_width(@title))/2 + @x, (@dy - @font.height)/2 + @y, @z, 1.0, 1.0, @color)
  end
end

File: src/common/text_button.py
import button
import pygame
from util import game_util
import zorder

class TextButton(button.Button):
    def __init__(self, owner, callback, publisher, x, y, label, font = 'hand.ttf', size = 10, color = (255, 255, 255), image_path = "default.png", layer = zorder.BUTTONS):
        font = pygame.font.Font(game_util.media(font), size)
        font = font.render(label, True, color)
        button.Button.__init__(self, owner, callback, publisher, x = x, y = y, image_path = image_path, layer = layer)
        glyph_rect = font.get_rect()
        self.image.blit(font, ((self.rect.width - glyph_rect.width)/2, (self.rect.height - glyph_rect.height)/2))
        
        
